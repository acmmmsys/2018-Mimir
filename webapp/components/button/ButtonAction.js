import React, { Component } from 'react';
import classnames from 'classnames';

import 'style/button/ButtonAction.scss';

const defaultClasses =Â [
    'link-button',
    'analysis-header-button',
];

class ButtonAction extends Component {

    static defaultProps = {
        classNames: [],
    }

    render() {
        let classNames = classnames(
            defaultClasses,
            this.props.classNames,
        );

        let label = this.props.children ? 
            this.props.children : this.props.label;

        return (
            <span 
                className={classNames}
                onClick={this.props.onClick}
            >
                {label}
            </span>
        );
    }
}

export default ButtonAction;