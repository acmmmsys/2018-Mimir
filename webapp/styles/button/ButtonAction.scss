@import "../colors";
@import "../mixins";
@import "~compass-mixins/lib/compass";

.analysis-header-button {
    margin-right: 20px;
}

.link-button {
    @include no-select;
    font-weight: 600;
    cursor: pointer;
    &:hover {
        color: $defaultHoverColor;
    }
}