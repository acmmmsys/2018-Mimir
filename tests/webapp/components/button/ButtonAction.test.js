import React from 'react';
import ButtonAction from '../../../../webapp/components/button/ButtonAction'
import renderer from 'react-test-renderer';

test('ButtonAction does render', () => {
  const wrapper = shallow(
      <ButtonAction>Hello Jest!</ButtonAction>
  );
  expect(wrapper).toMatchSnapshot();
});
