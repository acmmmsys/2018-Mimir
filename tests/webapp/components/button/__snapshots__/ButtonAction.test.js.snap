// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAction does render 1`] = `
<span
  className="link-button analysis-header-button "
>
  Hello Jest!
</span>
`;
